src/app/components/control-bar/control-bar.component.ts

```diff
@@ -14,8 +16,41 @@ export class ControlBarComponent implements OnInit {
   ngOnInit() {
   }
 
-  onClickOpenSearch() {
+  openSearchPage() {
     this.store.dispatch(new SetShowSearchAction({isShowSearch: true}));
   }
 
+  startNavigation() {
+    this.store.dispatch(new SetActiveNavAction({isActiveNav: true}));
+    this.store.pipe(
```

Comment:
the pipe should be on class scope not in function
Subscription should be removed on component removal
Configuration/button whether to use simulator extension